mpt_api_client/resources/catalog/products.py

```diff
@@ -56,38 +62,71 @@ class ProductsServiceConfig:
 
 
 class ProductsService(
+    CreateWithIconMixin[Product],
+    UpdateWithIconMixin[Product],
     PublishableMixin[Product],
-    ModifiableResourceMixin[Product],
+    GetMixin[Product],
+    DeleteMixin,
     CollectionMixin[Product],
     Service[Product],
     ProductsServiceConfig,
 ):
     """Products service."""
 
+    @override
     def create(
         self,
         resource_data: ResourceData,
         icon: FileTypes,
+        data_key: str = "product",
+        icon_key: str = "icon",
     ) -> Product:
         """Create product with icon.
 
         Args:
             resource_data: Product data.
             icon: Icon image in jpg, png, GIF, etc.
+            data_key: Key for the product data.
+            icon_key: Key for the icon.
 
         Returns:
             Created resource.
         """
-        files: dict[str, FileTypes] = {}
-        files["product"] = (
-            None,
-            json.dumps(resource_data),
-            "application/json",
+        return super().create(
```

Comment:
Didn't get here, what is the purpuse to override the basic method here? As I see you just run the same and pass the same set of parameters